File: swift/Sources/MasterMind/Guess.swift
//
//  Guess.swift
//  MasterMind
//
//  Created by oglimmer on 11/12/21.
//

import Foundation

class Guess {
    
    var list: Array<Color> = Array()
    
    func get(pos: Int) -> Color {
        return list[pos]
    }
    
    func parse(line: String) {
        let arr = line.split(separator: " ")
        list = arr
            .map({ Array($0)[0] })
            .filter({Color.isValid(c: $0)})
            .map({Color.get(character: $0)})
    }
    
    func incomplete() -> Bool {
        return list.count != 4
    }
    
}

File: js_oo/Guess.js

import { Color } from './Color.js'

export class Guess {

    list = Array()
    
    get(pos) {
        return this.list[pos]
    }
    
    parse(line) {
        const arr = line.split(" ")
        this.list = arr
            .map(e => e[0])
            .filter(e => Color.isValid(e))
            .map(e => Color.get(e))

    }
    
    incomplete() {
        return this.list.length != 4
    }
    
}